### PR TITLE
Improvement for AtomicGuard

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/core/src/atomic_guard.hpp [hunk 1]
```diff
@@ -14,6 +16,7 @@ class AtomicGuard {
         bool exp = false;
         while (m_atomic.load(std::memory_order_relaxed) || !m_atomic.compare_exchange_strong(exp, true)) {
             exp = false;
+            std::this_thread::yield();
```
Comment (nosovmik) [inline src/core/src/atomic_guard.hpp:19]: Does it fix some issue? I had a comment to recent changes on line 17, which was not still fixed: https://github.com/openvinotoolkit/openvino/pull/9473/files#r782254352  Is this fix somehow related to this?

Comment (ilyachur) [reply src/core/src/atomic_guard.hpp:19]: This line just switch thread in order to avoid several iterations of this cycle while thread context isn't switched.

**Review by nosovmik — APPROVED** [review]
